### ISSUE TITLE
Add docs for ProgressiveListView

### ISSUE THREAD

**Issue by osdnk** [Flipkart/recyclerlistview#464]
This class looks pretty nice and I'm using it in my app. Can we document it somehow?

**Comment by naqvitalha** [comment]
Yeah we will. We are planning to fix documentation soon. Can you explain me your use case btw? Internally we use it to speed up load times of very complex pages.

@manishPatwari

**Comment by osdnk** [comment]
I have the same case :) 
I need a list to show something a be responsive quickly and then load the rest of a list of complex items

**Comment by DaveLomber** [comment]
Also interested in it

@naqvitalha 
I found you use ProgressiveListView in FlashList
https://github.com/Shopify/flash-list/blob/218f314e63806b4fe926741ef73f8b9cd6ebc7eb/src/FlashList.tsx#L327

Would be great to understand benefits of ProgressiveListView against RecyclerListView
